app/javascript/gamesFactory.js

```diff
@@ -735,40 +831,84 @@ angular.module('dappChess').factory('games', function (crypto, navigation, gameS
-    } else {
-      let game = games.getGame(data.args.gameId);
-      if (typeof game === 'undefined') {
-        return;
-      }
-      game.timeoutStarted = data.args.timeoutStarted.toNumber();
-      game.timeoutState = data.args.timeoutState.toNumber();
-
-      /*
-       * -2 draw offered by nextPlayer
-       * -1 draw offered by waiting player
-       * 0 nothing
-       * 1 checkmate
-       * 2 timeout
-       */
-      if((game.timeoutState === -1 && game.nextPlayer === game.self.accountId) ||
-        (game.timeoutState === -2 && game.nextPlayer === game.opponent.accountId)) {
-        $rootScope.$broadcast('message',
-          'Player ' + game.opponent.username + ' wants to offer a draw',
-          'message', 'playgame-' + game.gameId);
-      }
-      if(game.timeoutState === 1 && game.nextPlayer === game.self.accountId) {
-        $rootScope.$broadcast('message',
-          'Player ' + game.opponent.username + ' claims that he won the game',
-          'message', 'playgame-' + game.gameId);
-      }
-      if(game.timeoutState === 2 && game.nextPlayer === game.self.accountId) {
-        $rootScope.$broadcast('message',
-          'Player ' + game.opponent.username + ' claims that he won the game due to a timeout',
-          'message', 'playgame-' + game.gameId);
+      console.error('error occured', err);
+      return;
+    }
+    let game = games.getGame(data.args.gameId);
+    if (typeof game === 'undefined') {
+      return;
+    }
+    game.timeoutStarted = data.args.timeoutStarted.toNumber();
+    game.timeoutState = data.args.timeoutState.toNumber();
+
+    /*
+     * -2 draw offered by nextPlayer
+     * -1 draw offered by waiting player
+     * 0 nothing
+     * 1 checkmate
+     * 2 timeout
+     */
+    if((game.timeoutState === -1 && game.nextPlayer === game.self.accountId) ||
+      (game.timeoutState === -2 && game.nextPlayer === game.opponent.accountId)) {
+      $rootScope.$broadcast('message',
+        'Player ' + game.opponent.username + ' wants to offer a draw',
+        'message', 'playgame-' + game.gameId);
+    }
+    if(game.timeoutState === 1 && game.nextPlayer === game.self.accountId) {
+      $rootScope.$broadcast('message',
+        'Player ' + game.opponent.username + ' claims that he won the game',
+        'message', 'playgame-' + game.gameId);
+    }
+    if(game.timeoutState === 2 && game.nextPlayer === game.self.accountId) {
+      $rootScope.$broadcast('message',
+        'Player ' + game.opponent.username + ' claims that he won the game due to a timeout',
+        'message', 'playgame-' + game.gameId);
+    }
+
+    $rootScope.$apply();
+
+    // Check own state to confirm or decline
+    if (data.args.timeoutState !== 0 && (
+        (game.chess.turn() === 'w' && game.self.color === 'white') ||
+        (game.chess.turn() === 'b' && game.self.color === 'black'))) {
+      if (
+          ([1, 2].indexOf(data.args.timeoutState) !== -1 && game.chess.in_checkmate()) || // jshint ignore:line
```

Comment:
Just for understanding: Why does this also apply to a timeout game status? Maybe you could also add this to the comments